lib/mayaUsd/undo/UsdUndoManager.cpp

```diff
@@ -31,7 +31,13 @@ UsdUndoManager& UsdUndoManager::instance()
 
 void UsdUndoManager::trackLayerStates(const SdfLayerHandle& layer)
 {
-    layer->SetStateDelegate(UsdUndoStateDelegate::New());
+    // Check if the layer has already been given a UsdUndoStateDelegate
+    // if the cast fails that means we need to set a new one.
+    auto usdUndoStateDelegatePtr
+        = TfDynamic_cast<UsdUndoStateDelegatePtr>(layer->GetStateDelegate());
+    if (!usdUndoStateDelegatePtr) {
+        layer->SetStateDelegate(UsdUndoStateDelegate::New());
+    }
```

Comment:
@ppt-adsk Thanks for great feedback.  Using a dynamic_cast to determine if a layer is already registered with our UsdUndoStateDelegate. If the cast fails, that means it needs to be set with a new one. The UsdUndoStateDelegate is stateless and doesn't need to be shared among layers.